core/git-access.c

```diff
@@ -551,7 +552,7 @@ int sync_with_remote(git_repository *repo, const char *remote, const char *branc
 	char *proxy_string;
 	git_config *conf;
 
-	if (prefs.git_local_only) {
+	if (is_local_cloud) {
```

Comment:
so you replace a global variable with a global variable. Why is that a good thing?

Reply:
Note that this does not exist anymore at the end of the series. At the end of the series this reads as
`if (state->is_cloud && !state->is_remote) {`
This is an artifact of "small" (yeah not really) commits.